src/packet/icmpv6.rs.in

```diff
@@ -127,4 +128,695 @@ pub mod Icmpv6Types {
+        /// |                                                               |
+        /// +                                                               +
+        /// |                                                               |
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// ```
+        ///
+        /// [RFC 4861 § 4.6.2]: https://tools.ietf.org/html/rfc4861#section-4.6.2
+        pub const PrefixInformation: NdpOptionType = NdpOptionType(3);
+
+        /// Redirected Header Option [RFC 4861 § 4.6.3]
+        ///
+        /// ```text
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// |     Type      |    Length     |            Reserved           |
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// |                           Reserved                            |
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// |                                                               |
+        /// ~                       IP header + data                        ~
+        /// |                                                               |
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// ```
+        ///
+        /// [RFC 4861 § 4.6.3]: https://tools.ietf.org/html/rfc4861#section-4.6.3
+        pub const RedirectedHeader: NdpOptionType = NdpOptionType(4);
+
+        /// MTU Option [RFC 4861 § 4.6.4]
+        ///
+        /// ```text
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// |     Type      |    Length     |           Reserved            |
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// |                              MTU                              |
+        /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        /// ```
+        ///
+        /// [RFC 4861 § 4.6.4]: https://tools.ietf.org/html/rfc4861#section-4.6.4
+        pub const MTU: NdpOptionType = NdpOptionType(5);
+    }
+
+    /// Neighbor Discovery Option [RFC 4861 § 4.6]
+    ///
+    /// ```text
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |     Type      |    Length     |              ...              |
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// ~                              ...                              ~
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// ```
+    ///
+    /// [RFC 4861 § 4.6]: https://tools.ietf.org/html/rfc4861#section-4.6
+    #[packet]
+    pub struct NdpOption {
+        #[construct_with(u8)]
+        option_type: NdpOptionType,
+        #[construct_with(u8)]
+        length: u8,
+        #[length_fn = "ndp_option_payload_length"]
+        #[payload]
+        data: Vec<u8>,
```

Comment:
Currently the options data is just kept as a `Vec<u8>`, but based on the options type we definitely know what the `Vec<u8>` should contain. I'd like to make this more user friendly, but that can (IMO should) be done in a future PR. Suggestions and design considerations on this would be welcomed.